pkg/operator/apiserver/controller/workload/workload.go

```diff
@@ -273,6 +273,34 @@ func (c *Controller) updateOperatorStatus(workload *appsv1.Deployment, operatorC
 		Status: operatorv1.ConditionFalse,
 	}
 
+	if !preconditionsReady {
+		message := "the operator didn't specify what preconditions are missing"
+		for _, err := range errs {
+			message = message + err.Error() + "\n"
+		}
+
+		// we are degraded, not available and we are not progressing
+
+		deploymentDegradedCondition.Status = operatorv1.ConditionTrue
+		deploymentDegradedCondition.Reason = "PreconditionNotReady"
+		deploymentDegradedCondition.Message = message
+
+		deploymentAvailableCondition.Status = operatorv1.ConditionFalse
+		deploymentAvailableCondition.Reason = "PreconditionNotReady"
+
+		deploymentProgressingCondition.Status = operatorv1.ConditionFalse
+		deploymentProgressingCondition.Reason = "PreconditionNotReady"
```

Comment:
how is progressing defined? Isn't waiting for preconditions a variant of progressing?

Reply:
```
// Progressing indicates that the operator is trying to transition the operand to a different state
OperatorStatusTypeProgressing = "Progressing"
```

when preconditions are not ready we are not trying to transition the operand (no progress).